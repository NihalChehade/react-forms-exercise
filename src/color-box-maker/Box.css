.button {
position: absolute;
right:5px;
top:0px;
width:0.1px;
color: rgb(6, 6, 6);
font-weight: bold;
background-color: transparent;
border: 0cap;
}

.Box {
  position:relative;
}