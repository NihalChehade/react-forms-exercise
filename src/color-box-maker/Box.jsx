import React from "react";
import './Box.css'
const Box = ({ id, color, width, height, removeBox }) => {
  return (
    <div>
     
        <div
          className="Box"
          style={{
            backgroundColor: color,
            width: `${width}px`,
            height: `${height}px`,
          }}
        ><button className='button' type="button"  onClick={() => removeBox(id)}>
        X
      </button></div>
     
      
        
      
    </div>
  );
};

export default Box;
