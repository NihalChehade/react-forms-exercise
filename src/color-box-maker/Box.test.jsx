import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Box from './Box';

// Smoke Test
test('renders Box component without crashing', () => {
  render(<Box id="1" color="red" width={100} height={100} removeBox={() => {}} />);
});

// Snapshot Test
test('matches snapshot', () => {
  const { asFragment } = render(<Box id="1" color="red" width={100} height={100} removeBox={() => {}} />);
  expect(asFragment()).toMatchSnapshot();
});

// Critical Business Logic Test
test('calls removeBox function with correct id when button is clicked', () => {
  const mockRemoveBox = vi.fn();
  render(<Box id="1" color="red" width={100} height={100} removeBox={mockRemoveBox} />);
  
  const button = screen.getByText('X');
  fireEvent.click(button);
  
  expect(mockRemoveBox).toHaveBeenCalledWith('1');
});
