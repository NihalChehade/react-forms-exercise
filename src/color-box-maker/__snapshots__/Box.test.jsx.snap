// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="Box"
      style="background-color: red; width: 100px; height: 100px;"
    >
      <button
        class="button"
        type="button"
      >
        X
      </button>
    </div>
  </div>
</DocumentFragment>
`;
